core/services/gateway/handlers/functions/allowlist/allowlist.go

```diff
@@ -210,7 +211,21 @@ func (a *onchainAllowlist) updateFromContractV1(ctx context.Context, blockNum *b
 	}
 
 	var allowedSenderList []common.Address
-	if !a.config.StoreAllowedSendersEnabled {
+	typeAndVersion, err := tosContract.TypeAndVersion(&bind.CallOpts{
+		Pending:     false,
+		BlockNumber: blockNum,
+		Context:     ctx,
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch the tos contract type and version")
+	}
+
+	version, err := extractContractVersion(typeAndVersion)
+	if err != nil {
+		return errors.Wrap(err, "failed to extract version")
+	}
+
+	if version < strings.ReplaceAll(tosContractMinBatchProcessingVersion, ".", "") {
```

Comment:
I just want to make sure that this string comparision is a safe way to determine version compatability.
Is this actually resiliant in all edge cases? Maybe we should extract this into a helper function such that we can properly unit test it? That would give me more confidence that there are no edge cases that could break this logic.

Reply:
sure I can extract it and create some unit tests for it 👍🏼